views/filters/date-part.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * nunjucks date-filter
+ * Not super complicated. Expects date in ISO8601 format YYYY-MM-DD.
+ * Splits it up into its parts, and returns the part being asked for.
+ */
+
+'use strict';
+
+var nunjucks = require('nunjucks');
+
+// a date part filter for Nunjucks
+// usage: {{ my_date | datePart(YYYY|MM|DD) }}
+function datePart(date, part) {
+  // [0] = YYYY
+  // [1] = MM
+  // [2] = DD
+  let parts = date.split('-');
```

Comment:
The input might need to be escaped before split. If you enter a dash, weird things happen.

![Screen Shot 2020-02-07 at 16 54 37](https://user-images.githubusercontent.com/950486/74070236-90398200-49ce-11ea-8ce3-30c7c8fef585.png)

![Screen Shot 2020-02-07 at 16 54 46](https://user-images.githubusercontent.com/950486/74070283-a5aeac00-49ce-11ea-949b-bb46e995ae76.png)

Reply:
Might be a good spot for a regex, enabling positive validation.

Reply:
@jeffmaher @dsamojlenko When something _unacceptable_ is entered into one of those boxes, is the same error message shown no matter what the user has done wrong? 
Also, is it able to identify which of the three fields has an invalid input?
Just thinking about having to write the error message for this.

Reply:
I've addressed this in two ways. First, I added some field validation to the markup so that the browser will not allow input other than numbers in these fields:

![image](https://user-images.githubusercontent.com/1187115/74164519-dde20480-4bf1-11ea-995d-9d498c71fb81.png)

Second, in case of shenanigans, on the server side, before combining the date parts, I pass them through a filter that only allows numbers:

![image](https://user-images.githubusercontent.com/1187115/74164599-01a54a80-4bf2-11ea-90e3-08bf0e0c82d7.png)

Reply:
@keithiopia When validating a date, we validate two things: each individual date part, and the combined date. 

If a date part is missing, we can throw a specific error (ie, Birthdate month is missing) - in those cases, we specify that error. If the date parts are all provided and that validation passes, then we combine the parts together and validate the whole date. 

In this case, we validate whether the date exists. Since we don't know what part of the date part fields caused the date to be invalid, we throw the error "Birthdate is not a valid date" and set focus to the first field in the set.

![image](https://user-images.githubusercontent.com/1187115/74165255-dbcc7580-4bf2-11ea-8ae9-d40d9b637248.png)